openproblems/tasks/cell_cell_communication/api.py

```diff
@@ -0,0 +1,90 @@
+from ...data.sample import load_sample_data
+from ...tools.decorators import dataset
+
+import numpy as np
+import pandas as pd
+import scanpy as sc
+
+
+def check_dataset(adata):
+    """Check that dataset output fits expected API."""
+    assert "label" in adata.obs
+    assert "ccc_target" in adata.uns
+    assert "response" in adata.uns["ccc_target"]
```

Comment:
Per the readme, looks like source, target, and ligand are also all required, no? Can we please test those?

Reply:
I would also like to see a test to ensure the sources and targets are valid celltypes:
```
assert np.all(np.isin(adata.uns["ccc_target"]["source"].unique(), adata.obs["celltype"].cat.categories))
assert np.all(np.isin(adata.uns["ccc_target"]["target"].unique(), adata.obs["celltype"].cat.categories))
```

Reply:
and that `ligand` and `response`, if not null, are valid gene names
```
assert np.all(np.isin(adata.uns["ccc_target"]["ligand"].unique(), adata.var.index))
if np.any(~pd.isna(adata.uns["ccc_target"]["receptor"])):
    assert np.all(np.isin(adata.uns["ccc_target"]["receptor"].unique(), adata.var.index))
```

Reply:
Re-reading the odds ratio code, I understand now -- the join is implicit based on which columns are available? So then it shouldn't be `nan` but we should do `if 'receptor' in adata.uns["ccc_target"].columns: ...`. Also from reading the examples, it sounds like `ligand` (allen) and `source` (tnbc) may also be missing?

Reply:
I understand now by the join why you want to allow it to be missing. Let's run the test if `if 'receptor' in adata.uns['ccc_target'].columns`

Reply:
OK, included these - with the difference that I don't force all entities (ligand/receptor) from `ccc_target` to be in `adata.var.index`, but rather at least some. For example, in my case I know that some ligand are not expected to match (e.g. you have a few ligands for which you can estimate an activity, but this ligand has a one-to-many mapping, and hence I implicitly don't want them to join). For this reason, I use `np.any`, as even though it's more promiscuous for the tests, such mismatches would already taken care of by the merge.

Reply:
I see. How are methods expected to know the names of these ligands? Is there a source of truth of ligand names in the NCBI database?

Reply:
Typically, each method comes with it's own set of 'prior-knowledge', but we harmonize these via liana (`liana::select_resource("Consensus")`), so that the methods are evaluated on the same resource. :)